src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Browser.cs

```diff
@@ -14,7 +14,11 @@ internal sealed class WebSocketHandle
         public WebSocket? WebSocket { get; private set; }
         public WebSocketState State => WebSocket?.State ?? _state;
 
-        public static ClientWebSocketOptions CreateDefaultOptions() => new ClientWebSocketOptions();
+        public static ClientWebSocketOptions CreateDefaultOptions()
+        {
+            BrowserWebSocket.ThrowIfPlatformNotSupported();
```

Comment:
Is this the right place for the throw? Which calls are you trying to protect?

Reply:
I would say yes and not protecting any calls that I know of.  It allows for checking in the tests if we are able to run the tests for browser or not.

Reply:
Then why not to do the checks in the tests and exclude the tests on configurations which don't support websockets?

Reply:
The tests check for PlatformNotSupported

Reply:
Also, the original code did the checks in the tests and it was suggested to do it this way.  Whichever way is ok with me just let me know.

Reply:
I don't think it makes any sense at this point to have WebSocketsSupported be true on v8 and then add another condition for when it is really true on browser.  If browser is the only platform that doesn't support WebSockets can't the condition can just become PlatformDetection.IsNotBrowser || PlatformDetection.IsBrowserDomSupported.  If there isn't a easy way to express that in attributes we can either stick it in InitWebSocketSupported or add another PlatformDetection value, something like IsClientNetworkingSupported which would also cover the outgoing http case.

Reply:
WebSocketsSupported is used other places so I moved the check to the tests only, if we ever intend to officially support wasm-browser on !browser we'll have more places to fix than just here.

Reply:
Thanks Larry